engine/apps/grafana_plugin/ui_url_builder.py

```diff
@@ -0,0 +1,50 @@
+import enum
+import typing
+from urllib.parse import urljoin
+
+from common.constants.plugin_ids import PluginID
+
+if typing.TYPE_CHECKING:
+    from apps.user_management.models import Organization
+
+
+class UIURLBuilder:
+    """
+    If `base_url` is passed into the constructor, it will override using `organization.grafana_url`
+    """
+
+    class OnCallPage(enum.StrEnum):
```

Comment:
The call signature for build url ends up being a bit long when you reference the enum, I think it would be nicer to use if we define these as functions which call build url instead of passing in str enums.  This way we can also enforce args like id when we want to.
```
Examples:
UIURLBuilder(org).alert_groups()
UIURLBuilder(org).alert_group_detail(id="abc") or ("abc")?
UIURLBuilder(org).settings(extra="?tab=ChatOps&chatOpsTab=Slack")
```

Reply:
good point 👍 addressed in https://github.com/grafana/oncall/pull/5137/commits/2570e24981589eff2e16817f0cf4dc59b8076e1e